modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -109,39 +115,50 @@ class GmosControllerEpics[T<:GmosController.SiteDependentTypes](encoders: GmosCo
     _ <- DC.setCcdYBinning(encode(dc.bi.y))
   } yield ()
 
-
+  private def smartSetParam[A: Equal](v: A, get: => Option[A], set: SeqAction[Unit]): SeqAction[Unit] =
+    get.flatMap(g => (g =/= v).option(set)).getOrElse(SeqAction.void)
 
   def setFilters(f: T#Filter): SeqAction[Unit] = {
     val (filter1, filter2) = encoders.filter.encode(f)
 
-    CC.setFilter1(filter1) *> CC.setFilter2(filter2)
+    smartSetParam(filter1, GmosEpics.instance.filter1, CC.setFilter1(filter1)) *>
```

Comment:
Will the `*>` make this run in parallel?

Reply:
No. It runs the left one, discards the result, and then runs the right one.

Reply:
If the first were to fail, would we know?

Reply:
Yes. Those are `EitherT[Task, SeqexecFailure, Unit]`. The `EitherT` shortcircuits on the left.